distribution/tools/plugin-cli/src/test/java/org/elasticsearch/plugins/cli/InstallPluginActionTests.java

```diff
@@ -404,6 +408,60 @@ public void testSomethingWorks() throws Exception {
         assertPlugin("fake", pluginDir, env.v2());
     }
 
+    public void testSlowSignatureVerificationMessage() throws Exception {
+        String url = "https://artifacts.elastic.co/downloads/elasticsearch-plugins/analysis-icu/analysis-icu-"
+            + Build.CURRENT.qualifiedVersion()
+            + ".zip";
+        final MessageDigest digest = MessageDigest.getInstance("SHA-512");
+
+        InstallPluginAction action = makeActionPluginThatDownloads(
+            "analysis-icu",
+            url,
+            null,
+            false,
+            ".sha512",
+            checksumAndFilename(digest, url),
+            newSecretKey(),
+            this::signature
+        );
+
+        final InstallPluginAction spied = spy(action);
+
+        // make the slow verification acceptable delay artificially low for testing
+        doReturn(100L).when(spied).acceptableSignatureVerificationDelay();
+
+        doAnswer(i -> {
+            Thread.sleep(200); // sleep 100ms more than acceptable delay
+            return null;
+        }).when(spied).computeSignatureForDownloadedPlugin(any(InputStream.class), any(InputStream.class), any(PGPSignature.class));
+
+        // NullPointerException expected because we are mocking the signature verification
+        expectThrows(NullPointerException.class, () -> installPlugin(new PluginDescriptor("analysis-icu", null), env.v1(), spied));
```

Comment:
That seems like a very generic exception to depend on, which could mask another problem (an actual NPE). Why would we not expect the install to succeed in this case? Can't we mock it in such a way that it works?

Reply:
Yeah valid point, I think I can make this work. Right now my mock method for computeSignatureForDownload is simply a sleep call, so the verification doesn't happen and a later call fails with NPE. But I can do the work of the original method too after the sleep and then it will succeed just fine.